pkg/backend-storage/backend-storage.go

```diff
@@ -65,11 +77,63 @@ func CreateIfNeeded(vmi *corev1.VirtualMachineInstance, clusterConfig *virtconfi
 			VolumeMode:       &modeFile,
 		},
 	}
+	// Whatever owns the VMI (i.e. a VM) now owns the PVC
+	for _, ownerRef := range vmi.ObjectMeta.OwnerReferences {
+		newRef := ownerRef.DeepCopy()
+		pvc.ObjectMeta.OwnerReferences = append(pvc.ObjectMeta.OwnerReferences, *newRef)
+	}
+
+	pvc, err = client.CoreV1().PersistentVolumeClaims(vmi.Namespace).Create(context.Background(), pvc, metav1.CreateOptions{})
+	if err != nil {
+		return "", err
+	}
+
+	return pvc.ObjectMeta.Name, nil
+}
 
-	_, err = client.CoreV1().PersistentVolumeClaims(vmi.Namespace).Create(context.Background(), pvc, metav1.CreateOptions{})
+func RemoveOldPVCFor(vmi string, pod *v1.Pod, client kubecli.KubevirtClient) {
+	if pod == nil {
+		return
+	}
+	toKeep := ""
+	for _, volume := range pod.Spec.Volumes {
+		if volume.PersistentVolumeClaim != nil {
+			toKeep = volume.PersistentVolumeClaim.ClaimName
+			break
+		}
+	}
+	if toKeep == "" {
+		return
+	}
+	pvcList, err := client.CoreV1().PersistentVolumeClaims(pod.Namespace).List(context.Background(), metav1.ListOptions{
+		LabelSelector: "kubevirt.io/vm=" + vmi,
+	})
 	if err != nil {
-		return false, err
+		return
+	}
+	for _, pvc := range pvcList.Items {
+		if pvc.Name != toKeep {
```

Comment:
What is the PVC that you want to keep? Are we deleting here also other PVCs? Like for example other RWX PVCs that the VMI might own?

Reply:
We want to keep the PVC that was created for the migration target.
The `List()` call is restricted to the label selector I created, however I need to change its name (`kubevirt.io/vm=`) to something more obvious.